src/data/services/LmsApiService.js

```diff
@@ -384,6 +385,57 @@ class LmsApiService {
     const url = `${LmsApiService.baseUrl}/enterprise/api/v1/analytics-summary/${enterpriseUUID}`;
     return LmsApiService.apiClient().post(url, formData);
   }
+
+  static updateUserActiveEnterprise = (enterpriseId) => {
+    const url = `${configuration.LMS_BASE_URL}/enterprise/select/active/`;
+    const formData = new FormData();
+    formData.append('enterprise', enterpriseId);
+
+    return LmsApiService.apiClient().post(
+      url,
+      formData,
+    );
+  };
+
+  static loginRefresh = async () => {
+    const loginRefreshUrl = `${configuration.LMS_BASE_URL}/login_refresh/`;
+
+    try {
+      return LmsApiService.apiClient().post(loginRefreshUrl);
+    } catch (error) {
+      const isUserUnauthenticated = error.response?.status === 401;
+      if (isUserUnauthenticated) {
+        // Clean up the cookie if it exists to eliminate any situation
+        // where the cookie is not expired but the jwt is expired.
+        const cookies = new Cookies();
+        cookies.remove(configuration.ACCESS_TOKEN_COOKIE_NAME);
+      }
+      return Promise.resolve();
+    }
+  };
+
+  static fetchEnterpriseLearnerData(options) {
+    const enterpriseLearnerUrl = `${configuration.LMS_BASE_URL}/enterprise/api/v1/enterprise-learner/`;
+    const queryParams = new URLSearchParams({
+      ...options,
+      page: 1,
+    });
+    const url = `${enterpriseLearnerUrl}?${queryParams.toString()}`;
+    return LmsApiService.apiClient().get(url);
+  }
+
+  static getActiveLinkedEnterprise(username) {
+    return this.fetchEnterpriseLearnerData({ username }).then((response) => {
+      const { data } = response;
+      const results = data?.results;
```

Comment:
[curious] When would `data` be undefined? Do we need to use the optional chaining you have here? If we do, when `data` is undefined, `results` would also be undefined and would subsequently throw a JS error on line 431 when trying to access `results.length`.

Reply:
I think I was mostly just following the pattern in the file- but I agree I can't think of a reason why it would be undefined.